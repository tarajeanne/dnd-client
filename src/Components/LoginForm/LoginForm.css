.LoginPage {
  display: block;
  margin: auto;
  max-width: 400px;
  border: 1px solid #7A6263;
  padding: 10px;
}

.login-input-area {
  text-align: center;
}


.login-label {
  display: inline-block;
  width: 100px;
  text-align: right;
  margin-right: 10px;
}

.login-submit {
  display: block;
  margin: 15px auto;
}