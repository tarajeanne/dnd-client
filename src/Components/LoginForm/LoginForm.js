import React from 'react';
import AuthApiService from '../../services/auth-api-service';
import PropTypes from 'prop-types';
import './LoginForm.css';

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {error: null};
  }

  static propTypes = {
    onLoginSuccess: PropTypes.func
  }

  static defaultProps = {
    onLoginSuccess: () => {}
  };

  handleSubmitJwtAuth = (e) => {
    e.preventDefault();
    this.setState({ error: null })
    const { username, password } = e.target;

    AuthApiService.postLogin({
      username: username.value,
      password: password.value
    })
    .then(res => {
      password.value = '';
      username.value = '';
      this.props.onLoginSuccess();
    })

    .catch(res => {
      this.setState({ error: res.error })
    });
  };

  render() {
    const { error } = this.state;
    return (
      <form className="LoginForm" onSubmit={this.handleSubmitJwtAuth}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="login-input-area">
          <label htmlFor="LoginForm__username" className="login-label">Username:</label>
          <input required className="login-input" name="username" id="LoginForm__username" />
        </div>
        <div className="login-input-area">
          <label htmlFor="LoginForm__password" className="login-label">Password:</label>
          <input
            required
            className="login-input"
            name="password"
            type="password"
            id="LoginForm__password"
          />
        </div>
        <button className="login-submit" type="submit">Login</button>
      </form>
    );
  }
}

export default LoginForm;