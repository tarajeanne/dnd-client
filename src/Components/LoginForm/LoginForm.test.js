import React from 'react';
import LoginForm from './LoginForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('LoginForm', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
