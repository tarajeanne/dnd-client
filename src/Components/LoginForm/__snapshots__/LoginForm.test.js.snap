// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="username"
  >
    <label
      htmlFor="LoginForm__username"
    >
      User name
    </label>
    <input
      id="LoginForm__username"
      name="username"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
